src/data/Analysis.js

```diff
@@ -31,32 +31,75 @@ export class Analysis {
   async fromZip(zipFiles) {
     // eslint-disable-next-line no-console
     const jsZip = new JSZip();
+      const BINARY = 1;
+      const TEXT = 2;
 
     for (let zipFile of zipFiles) {
-      const zip = await jsZip.loadAsync(zipFile.binary);
 
-      let files = Object.keys(zip.files).filter((name) =>
-        name.endsWith('1/1r'),
-      );
+      let files = {
+        ser: BINARY,
+        fid: BINARY,
+        acqus: TEXT,
+        acqu2s: TEXT,
+        procs: TEXT,
+        proc2s: TEXT,
+        '1r': BINARY,
+        '1i': BINARY,
+        '2rr': BINARY,
+      };
 
-      // console.log(this.spectra);
-      // eslint-disable-next-line no-await-in-loop
-      files = await Promise.all(
-        files.map(async (file) => {
+      const zip = await jsZip.loadAsync(zipFile.binary);
+
+      let folders = zip.filter(function(relativePath) {
```

Comment:
Please use arrow function

Reply:
seems to me that what you want to do is simply:
```suggestion
     let folders = zip.filter( (path) => path.match( /(ser|fid|1r|2rr)$/ ));
```